### PR TITLE
Add __pos__ for Parameter

### PR DESCRIPTION
**Pull request by eendebakpt**
### Summary

Add `__pos__` for `Parameter` and `ParameterExpression`

### Details and comments

The implementation is simular to the `__neg__`

### CODE REVIEW

File: qiskit/circuit/parameterexpression.py [hunk 1]
```diff
@@ -327,6 +327,9 @@ def __rsub__(self, other):
     def __mul__(self, other):
         return self._apply_operation(operator.mul, other)
 
+    def _pos__(self):
```
Comment (jakelishman) [inline qiskit/circuit/parameterexpression.py]:
```suggestion
    def __pos__(self):
```

**Review by jakelishman — COMMENTED** [review]
This seems sensible in spirit to me, thanks.

I guess it would have made more sense if we were able to use `operator.pos` in the `_apply_operation`, but that function seems to be set up to assume binary operations, so never mind on that front.

I'm not wild about introducing a floating-point number with this, though, and the same goes for `__neg__`.  Definitely feels like we might at least want to make them multiplication by _integers_ rather than floats (if we can't just use `operator.pos` and `operator.neg` directly) to avoid introducing imprecisions into the evaluation contexts.

Please could you also add a circuits feature release note about this?

**Comment by eendebakpt** [comment]
> This seems sensible in spirit to me, thanks.
> 
> I guess it would have made more sense if we were able to use `operator.pos` in the `_apply_operation`, but that function seems to be set up to assume binary operations, so never mind on that front.
> 
> I'm not wild about introducing a floating-point number with this, though, and the same goes for `__neg__`. Definitely feels like we might at least want to make them multiplication by _integers_ rather than floats (if we can't just use `operator.pos` and `operator.neg` directly) to avoid introducing imprecisions into the evaluation contexts.
> 
> Please could you also add a circuits feature release note about this?

Multiplication with an int seems nicer indeed. I can change it for this PR, but I am not sure whether there are any implications for `__neg__`. Let me know

**Comment by jakelishman** [comment]
Let's make at least `__pos__` be multiplication by integer here. I think we ought to change `__neg__` too - if the tests don't complain about it (and hopefully nothing in our test suite depends on that questionable behaviour...), but happy to do that in a follow-up if you prefer.

**Review by jakelishman — APPROVED** [review]
Ace, thanks!

**Pull request merged**